File: app/src/main/java/ua/com/foxminded/locationtrackera/ui/tracker/state/TrackerScreenEffect.java
package ua.com.foxminded.locationtrackera.ui.tracker.state;

import ua.com.foxminded.locationtrackera.mvi.states.AbstractEffect;
import ua.com.foxminded.locationtrackera.ui.tracker.TrackerContract;

public abstract class TrackerScreenEffect extends AbstractEffect<TrackerContract.View> {

    public static class Logout extends TrackerScreenEffect {
        @Override
        public void handle(TrackerContract.View screen) {
            screen.proceedToSplashScreen();
        }
    }

    public static class ShowDialogFragment extends TrackerScreenEffect {

        public final int argType;
        public final int message;
        public final int negativeButton;
        public final int positiveButton;

        public ShowDialogFragment(int argType, int message, int negativeButton, int positiveButton) {
            this.argType = argType;
            this.message = message;
            this.negativeButton = negativeButton;
            this.positiveButton = positiveButton;
        }

        @Override
        public void handle(TrackerContract.View screen) {
            screen.showDialogFragment(argType, message, negativeButton, positiveButton);
        }
    }
}

File: app/src/main/java/ua/com/foxminded/locationtrackera/ui/tracker/state/TrackerScreenEffect.kt
package ua.com.foxminded.locationtrackera.ui.tracker.state

import ua.com.foxminded.locationtrackera.ui.tracker.TrackerContract
import ua.com.foxminded.locationtrackera.mvi.states.AbstractEffect

abstract class TrackerScreenEffect : AbstractEffect<TrackerContract.View>() {

    class Logout : TrackerScreenEffect() {
        override fun handle(screen: TrackerContract.View) {
            screen.proceedToSplashScreen()
        }
    }

    class ShowDialogFragment(
        val argType: Int,
        val message: Int,
        val negativeButton: Int,
        val positiveButton: Int
    ) : TrackerScreenEffect() {

        override fun handle(screen: TrackerContract.View) {
            screen.showDialogFragment(argType, message, negativeButton, positiveButton)
        }
    }
}